### PR TITLE
chore: upgrade testing library

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

we are on testing-library for react 17.
So we have logs in tests saying that render is done with the compatibility layer.

```
Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot
```

**What is the chosen solution to this problem?**

upgrade testing-library:
* [x] build
* [ ] test

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Gbacc — APPROVED** [review]
Won't review everything one by one but it seems good overall

File: packages/components/src/wrap.test.tsx [hunk 1]
```diff
@@ -9,14 +8,16 @@ describe('wrap', () => {
 	);
 	Button.displayName = 'Button';
 	Button.foo = 'bar';
-	it('should create a component', () => {
+	it('should create a component', async () => {
+		const user = userEvent.setup();
```
Comment (smouillour) [inline packages/components/src/wrap.test.tsx:12]: fix lint issue:
Property 'setup' does not exist on type '{ click: (element: Element, init?: MouseEventInit | undefined, { skipHover, clickCount, skipPointerEventsCheck, }?: (clickOptions & PointerOptions) | undefined) => void; ... 10 more ...; keyboard: { ...; }; }'.ts(2339)

File: packages/a11y/src/Gesture/preventScroll.test.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { render, screen } from '@testing-library/react';
+import { preventScroll } from './preventScroll';
+import userEvent from '@testing-library/user-event';
+
+describe('preventScroll', () => {
+	it('Prevent arrow keys', async () => {
+		const user = userEvent.setup();
```
[the remark this answers is not in the document]
Comment (jmainguytalend) [reply packages/a11y/src/Gesture/preventScroll.test.tsx:7]: I don't have this problem. Maybe you didn't do `yarn install` ?

File: packages/a11y/tsconfig.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "extends": "@talend/scripts-config-typescript/tsconfig.json",
   "include": ["src/**/*"],
   "compilerOptions": {
+    "types": ["jest", "@testing-library/jest-dom"],
```
[the remark this answers is not in the document]
Comment (jmainguytalend) [reply packages/a11y/tsconfig.json:5]: Fixed. Thank you

File: packages/a11y/src/Gesture/withMonthCalendarGesture.test.tsx [hunk 4]
```diff
@@ -39,120 +39,145 @@ describe('withMonthCalendarGesture', () => {
 	});
 	describe('LEFT keydown', () => {
 		it('should focus on previous month in the same set of months', async () => {
+			const user = userEvent.setup();
```
[the remark this answers is not in the document]
Comment (jmainguytalend) [reply packages/a11y/src/Gesture/withMonthCalendarGesture.test.tsx:42]: Same as previous comment

File: packages/components/src/GridLayout/Tile/Tile.component.test.tsx [hunk 5]
```diff
@@ -22,15 +22,17 @@ function Consumer() {
 }
 
 describe('Grid tiles', () => {
-	it('should render children and setup context', () => {
+	it('should render children and setup context', async () => {
+		const user = userEvent.setup();
```
[the remark this answers is not in the document]
Comment (jmainguytalend) [reply packages/components/src/GridLayout/Tile/Tile.component.test.tsx:26]: Same thing than for preventScroll.test.tsx

File: packages/components/src/wrap.test.tsx (hunk 1 above)
Comment (jmainguytalend) [reply packages/components/src/wrap.test.tsx:12]: Same thing than for preventScroll.test.tsx

**Review by smouillour — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**